libbeat/publisher/pipeline/client_test.go

```diff
@@ -276,65 +286,251 @@ func TestClientWaitClose(t *testing.T) {
 }
 
 func TestMonitoring(t *testing.T) {
-	const (
-		maxEvents  = 123
-		batchSize  = 456
-		numClients = 42
-	)
-	var config Config
-	err := conf.MustNewConfigFrom(map[string]interface{}{
-		"queue.mem.events":           maxEvents,
-		"queue.mem.flush.min_events": 1,
-	}).Unpack(&config)
-	require.NoError(t, err)
+	t.Run("output metrics", func(t *testing.T) {
+		const (
+			maxEvents  = 123
+			batchSize  = 456
+			numClients = 42
+		)
+		var config Config
+		err := conf.MustNewConfigFrom(map[string]interface{}{
+			"queue.mem.events":           maxEvents,
+			"queue.mem.flush.min_events": 1,
+		}).Unpack(&config)
+		require.NoError(t, err)
+
+		metrics := monitoring.NewRegistry()
+		telemetry := monitoring.NewRegistry()
+
+		beatInfo := beat.Info{}
+		beatInfo.Monitoring.Namespace = monitoring.GetNamespace("TestMonitoring.outputMetrics")
+		pipeline, err := Load(
+			beatInfo,
+			Monitors{
+				Metrics:   metrics,
+				Telemetry: telemetry,
+			},
+			config,
+			processing.Supporter(nil),
+			func(outputs.Observer) (string, outputs.Group, error) {
+				clients := make([]outputs.Client, numClients)
+				for i := range clients {
+					clients[i] = newMockClient(func(publisher.Batch) error {
+						return nil
+					})
+				}
+				return "output_name", outputs.Group{
+					BatchSize: batchSize,
+					Clients:   clients,
+				}, nil
+			},
+		)
+
+		require.NoError(t, err)
+		defer pipeline.Close()
+
+		telemetrySnapshot := monitoring.CollectFlatSnapshot(telemetry, monitoring.Full, true)
+		metricsSnapshot := monitoring.CollectStructSnapshot(metrics, monitoring.Full, true)
+		_ = metricsSnapshot
```

Comment:
I'm confused here, what is the goal of `metricsSnapshot` if it's not used?

Was it left over from a test/debug?